### ISSUE TITLE
Error when editing enum array columns with default value

### ISSUE THREAD

**Issue by hallsbyra** [dbeaver/dbeaver#9828]
#### System information: 
- Windows 10 x64 Latest
- DBeaver 7.2.1

#### Connection specification:
- PostgreSQL 12.4
- PostgreSQL

#### Describe the problem you're observing:

Cannot add/remove array elements in table data view for column with data type enum array with default value.

#### Steps to reproduce, if exist:

1. Create test table:

```sql
create type "test_enum" as enum('A', 'B', 'C');
create table "test_table" ("id" character varying not null, "val" "test_enum" array NOT NULL DEFAULT array[]::test_enum[], primary key("id"));
```

2. Open data grid view for `test_table`.
3. Create a new row.
4. Type any value in the `id` column.
5. Save the record.
6. Refresh the record to load the default value `{}` in the `val` column.
7. Select the `val` column and try to add a couple of array elements.
8. `NULL` is displayed instead of the available enum values.

![image](https://user-images.githubusercontent.com/2835896/93762992-e7642e00-fc10-11ea-9897-b5cb6404a6ab.png)

9. Trying to save the record (even with empty array) results in the following error: 

> Error synchronizing data with database
> 
> Reason:
> Error creating struct

**NB** - The problem is related to the default value of the column. If the column is created without a default value, there is no problem:
```sql
create table "test_table" ("id" character varying not null, "val" "test_enum" array, primary key("id"));
```

**Comment by kseniiaguzeeva** [comment]
Thank you for the report.

**Comment by alex88** [comment]
I just wanted to report that I have the same issue trying to edit a `ARRAY[]::character` column.

Rarely it does work but most of the time it doesn't

**Comment by shawnngtq** [comment]
I am currently using dbeaver 7.2.2 on my ubuntu OS and notice the same problem `Error creating struct` for my PostgreSQL 12.4.

For `_varchar` and `_numeric` datatype when I `import data`

**Comment by hkorsvoll** [comment]
I am experiencing the same error. I can add some details. If you open the array data type of test_enum (_test_enom). You will see the source of this type is:
```
CREATE TYPE _test_enum (
	INPUT = array_in,
	OUTPUT = array_out,
	RECEIVE = array_recv,
	SEND = array_send,
	ANALYZE = array_typanalyze,
	ALIGNMENT = 4,
	STORAGE = any,
	CATEGORY = A,
	ELEMENT = test_enum,
	DELIMITER = ',');
```
The element is referencing to the enum type as expected. But if you open the property tab, you see that the Array element is not populated.
![bilde](https://user-images.githubusercontent.com/973301/97292500-9c5ebb80-184b-11eb-823c-beadfc236d35.png)

**Comment by hkorsvoll** [comment]
This looks like the same as #9965

**Comment by akilovich** [comment]
It appears that the issue was fixed in 7.2.4.

**Comment by hallsbyra** [comment]
Works fine, thanks!
